### PR TITLE
Block Autocompleter: Force icon color to text color when item is selected

### PR DESCRIPTION
**Pull request by t-hamano**
Follow up #60131

## What?

This PR overrides the icon color in the Block Autocompleter popover when an item is selected, increasing icon visibility.

| Before | After |
|--------|--------|
| image | image | 

## Why?

#60131 changed the background color of selected items to the primary color. In the block autocompleter, icons are rendered via the `BlockIcon` component. This component references the `foreground` and `background` of the icon prop and [injects its color as inline styles](https://github.com/WordPress/gutenberg/blob/c56672504bada434aee1ec331366cd2a10ab2b38/packages/block-editor/src/components/block-icon/index.js#L23-L32). As a result, the icons are always the same color, which can reduce visibility.

## How?

When the button is selected, overrides the color of the block icon component with `inherit`. This will inherit the appropriate color and match the button text color.

## Testing Instructions

- On the block editor, use the slash inserter to find blocks with icon colors (WordPress, YouTube, Spotify, etc.)
- When the button is selected, the icon should appear white, along with the text.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/f22e5a33-b234-4976-8fc3-c082dc208693

### CODE REVIEW

**Comment by t-hamano** [comment]
P.S. We might want to apply this approach to the list view as well.

image

File: packages/block-editor/src/autocompleters/style.scss [hunk 1]
```diff
@@ -4,6 +4,10 @@
 	.block-editor-block-icon {
 		margin-right: $grid-unit-10;
 	}
+
+	&.is-selected .block-editor-block-icon {
+		color: inherit !important;
```
Comment (ellatrix) [inline packages/block-editor/src/autocompleters/style.scss:9]: Was there a way to choose between coloured or block and white icons? Should we always choose black and white for autocomplete maybe?

**Review by ellatrix — APPROVED** [review]

File: packages/block-editor/src/autocompleters/style.scss (hunk 1 above)
Comment (jasmussen) [reply packages/block-editor/src/autocompleters/style.scss:9]: Even `!important`'s can be overridden by plugins if they want it enough — at least until we can shadow-DOM these components :) 

I'd tend to agree that color for blocks should be enforced in this view, it's a way to enforce contrast.

Comment (t-hamano) [reply packages/block-editor/src/autocompleters/style.scss:9]: > Was there a way to choose between coloured or block and white icons? Should we always choose black and white for autocomplete maybe?

If we choose white/black, I think all we need to do is delete `showColors` [here](https://github.com/WordPress/gutenberg/blob/3d8b192c3df3b02f4c1efc9d8154b715dc189600/packages/block-editor/src/autocompleters/block.js#L103).

image

In my experience, some block plugins color their icons to highlight their branding. Therefore, I think it is better to respect the colored icon when the block is not selected.

**Review by tyxla — APPROVED** [review]
This makes sense to me and works as expected 👍

**Comment by jasmussen** [comment]
Nice one. Thank you!

**Comment by t-hamano** [comment]
I found another issue with the text color of this component, so I submitted #64294.

It will conflict with this PR, so I will update this PR and retest after #64294 is merged.

**Comment by t-hamano** [comment]
Updated this PR since #64294 was merged. In #64294, the `is-selected` CSS class was kept only for backwards compatibility, so in this PR I also decided to rely on the `aria-selected` attribute instead of `.is-selected` CSS class. I think it's working as expected:

https://github.com/user-attachments/assets/b6e6c0b1-6336-41be-bc33-6b40c8ac9b66

**Review by tyxla — APPROVED** [review]
Works well, @t-hamano - thanks for the fix! 🚀

**Pull request merged**